docs/docs/deploying-to-cloudflare-workers.md

```diff
@@ -0,0 +1,127 @@
+
+This guide will get you started in a few steps:
+
+## Step 1: Installing wrangler CLI
+
+Workers Sites requires [wrangler](https://developers.cloudflare.com/workers/tooling/wrangler/). The more straight forward way to install wrangler is with [npm](https://www.npmjs.com/), run the following command:
+
+```shell
+npm i -g @cloudflare/wrangler
+```
+
+## Step 2: Init Project
+
+To create the Worker code that will serve your Gatsby files, from the root of your Gatsby project run:
+
+```
+wrangler init --site
+```
+
+You'll notice your project structure should now look something like:
+
+```
+├── workers-site
+└── wrangler.toml
+│   ├── index.js
+│   ├── package-lock.json
+│   └── package.json
+├── public
+│   ├── 404
+│   ├── 404.html
+    ..
+├── src
+│   ├── components
+│   ├── images
+│   └── pages
+├── gatsby-config.js
+├── package.json
+```
+
+## Configure
+
+To authenticate into your Cloudflare account run:
+
+```
+wrangler config
+```
+
+Follow the [Quick Start](https://developers.cloudflare.com/workers/quickstart/#configure) for steps on gathering the correct account ID and API token to link wrangler to your Cloudflare account.
+
+If you don't already have a workers.dev domain run:
+
+```
+wrangler subdomain
+```
+
+Then, add to your `wrangler.toml` the account ID and set `bucket` to the Gatsby files:
+
+```
+name = "gatsby-project"
+type = "webpack"
+account_id = "abcd..."
+workers_dev = true
+
+[site]
+bucket = "./public"
+entry-point = "workers-site"
+```
+
+This deploys to your workers.dev subdomain. For a custom domain see [Quick Start]().
+
+## Step 3: Deploy
+
+You can deploy your application by running the following command in the root of the project directory:
+
+```shell
+wrangler publish
+```
+
+Now your site is available at gatsby-project.subdomain.workers.dev!
+
+## Step 4: CI with Github Actions
+
+Use wrangler's Github action [plugin](https://github.com/cloudflare/wrangler-action) to automatically deploy to Workers every time you push to master.
+
+Once Github actions is enabled on your repo add a file to projects root called `.github/workflows/main.yml` with the contents:
+
+```
+name: Deploy production site
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Navigate to repo
+      run: cd $GITHUB_WORKSPACE
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
+    - name: Install deps
+      run: npm install
+    - name: Build docs
+      run: npm run build
+    - name: Publish
+      uses: signalnerve/wrangler-action@1.0.0
+      with:
+        apiKey: ${{ secrets.CF_API_KEY }}
```

Comment:
We should swap this usage out with API tokens -

```
with:
  apiToken: ${{ secrets.CF_API_TOKEN }}
```

and update the below section on where to find it accordingly